js/modules/k6/execution/execution.go

```diff
@@ -0,0 +1,169 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2021 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package execution
+
+import (
+	"context"
+	"errors"
+
+	"github.com/dop251/goja"
+
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/lib"
+)
+
+// Execution is a JS module to return information about the execution in progress.
+type Execution struct{}
+
+// New returns a pointer to a new Execution.
+func New() *Execution {
+	return &Execution{}
+}
+
+// GetVUStats returns information about the currently executing VU.
+func (e *Execution) GetVUStats(ctx context.Context) (goja.Value, error) {
+	vuState := lib.GetState(ctx)
+	if vuState == nil {
+		return nil, errors.New("getting VU information in the init context is not supported")
+	}
+
+	rt := common.GetRuntime(ctx)
+	if rt == nil {
+		return nil, errors.New("goja runtime is nil in context")
+	}
+
+	stats := map[string]interface{}{
+		"id":         vuState.VUID,
+		"idGlobal":   vuState.VUIDGlobal,
+		"idScenario": vuState.VUIDScenario,
+		"iteration":  vuState.Iteration,
+		"iterationScenario": func() goja.Value {
+			return rt.ToValue(vuState.GetScenarioVUIter())
+		},
+	}
+
+	obj, err := newLazyJSObject(rt, stats)
+	if err != nil {
+		return nil, err
+	}
+
+	return obj, nil
+}
+
+// GetScenarioStats returns information about the currently executing scenario.
+func (e *Execution) GetScenarioStats(ctx context.Context) (goja.Value, error) {
+	ss := lib.GetScenarioState(ctx)
+	vuState := lib.GetState(ctx)
+	if ss == nil || vuState == nil {
+		return nil, errors.New("getting scenario information in the init context is not supported")
+	}
+
+	rt := common.GetRuntime(ctx)
+	if rt == nil {
+		return nil, errors.New("goja runtime is nil in context")
+	}
+
+	var iterGlobal interface{}
+	if vuState.GetScenarioGlobalVUIter != nil {
+		iterGlobal = vuState.GetScenarioGlobalVUIter()
+	} else {
+		iterGlobal = goja.Null()
+	}
+
+	stats := map[string]interface{}{
+		"name":      ss.Name,
+		"executor":  ss.Executor,
+		"startTime": ss.StartTime,
```

Comment:
This probably needs a type conversion, either to a proper string or to some sort of a unix timestamp (with milliseconds). Right now, it's Go's `time.Time` type and behaves very strangely in JS. For example:
```js
export default function () {
    let scenarioStats = execution.getScenarioStats();
    console.log(scenarioStats.startTime);
    console.log(JSON.stringify(scenarioStats.startTime));
    console.log(typeof scenarioStats.startTime);
    console.log(scenarioStats.startTime instanceof Date);
    console.log(typeof scenarioStats.startTime.getMonth);
    console.log(Object.keys(scenarioStats.startTime));
}
```

will result in this:
```
2021-06-17 16:06:17.00787938 +0300 EEST m=+0.488634128
"2021-06-17T16:06:17.00787938+03:00"
object
false
undefined
add,addDate,after,appendFormat,before,clock,date,day,equal,format,gobEncode,hour,iSOWeek,in,isZero,local,location,marshalBinary,marshalJSON,marshalText,minute,month,nanosecond,round,second,string,sub,truncate,uTC,unix,unixNano,weekday,year,yearDay,zone
```

Reply:
Yeah, I ran into this when I was writing the tests, see https://github.com/dop251/goja/issues/170.

I'm now converting it in Go, see 80671dfa.

Reply:
hm, given the fact Date doesn't have timezone and I am not really certain what anyone will use the Date Object instead of a the seconds since UTC epoch :thinking: I am somewhat for returning it as a number to be honest

Reply:
You're right, fixed in f32a4d05.